src/modules/CaaSMapper.ts

```diff
@@ -298,6 +298,35 @@ export class CaaSMapper {
     }
   }
 
+  async mapLinksInRichTextElements(richTextElements: RichTextElement[], path: NestedPath) {
```

Comment:
This whole mapping function seems kind of unnecessary to me since the needed properties here are `type: 'Link'` and `data: ...`.
By putting a Link object in RickTextElement's data prop, everything gets pushed 1 layer deeper which makes things more complicated than they need to be in my opinion.

Since `link` is already a valid type of RichTextElement, couldn't we simply map the link props directly into the structure of RichTextElement?

Reply:
Note: Discuss if reducing the depth by one layer here by merging the `Link` object in the `RichTextElement` is a good idea. Probably not, but maybe yes.

Reply:
As discussed in weekly, for now we can't reduce the complexity here.